Lib/test/support/__init__.py

```diff
@@ -467,15 +468,23 @@ def requires_debug_ranges(reason='requires co_positions / debug_ranges'):
 else:
     unix_shell = None
 
-# wasm32-emscripten is POSIX-like but does not provide a
-# working fork() or subprocess API.
+# wasm32-emscripten and -wasi are POSIX-like but do not
+# have subprocess or fork support.
 is_emscripten = sys.platform == "emscripten"
+is_wasi = sys.platform == "wasi"
 
-has_fork_support = hasattr(os, "fork") and not is_emscripten
+has_fork_support = hasattr(os, "fork") and not is_emscripten and not is_wasi
```

Comment:
[meta comment] weird, so those platforms have os.fork but it cannot be used? unusual...

Reply:
Emscripten and WASI are weird platforms. They do their best to mimic a Linux-like, mostly POSIX compatible platform with MUSL libc. The libc has a ``fork`` function, but the "operating system" (JavaScript glue code or WASI runtime) don't support the fork call. The syscall always fails with ENOTSUP.

Reply:
we could ponder disabling it in that situation, but the code could run on a better runtime that supports it (like running on a more recent kernel) so i guess this is just life for now.  thanks!